### DISCUSSION TITLE
File association VS Code

### DISCUSSION THREAD

**Discussion by Ruslan618** [clangd/clangd#1273] [Q&A] [answered]
Hi everyone, thanks to the developers for a great app. Only I have a problem and I will be glad for any help.
I am programming in MQL, it is very similar with c++, so I use file association. 
   "files.associations": {
      " *.mq4": "cpp",
      " *.mqh": "cpp"
  },
But it does not help, or rather the file is recognized as C++, but nothing works – outline, vars, methods etc.
But if I change .mq4 in my file to .cpp everything works immediately.
So, it can somehow be possible to make my mql files work as a cpp so that clangd processes them properly.

**Comment by HighCommander4** [comment]
Modifying your compile_commands.json to add `-x c++` to commands for these files should help.

**Reply by Ruslan618** [reply]
Can you plz write in full what it should look like.

**Reply by HighCommander4** [reply]
@i-ky's suggestion is probably the easier one.

To get the flag into the `compile_commands.json`, you'd probably want to add it to your build system (e.g. to your `makefile`) and re-generate the `compile_commands.json`. See https://clangd.llvm.org/installation#project-setup for some options of how to generate a `compile_commands.json` for your project.

**Reply by Ruslan618** [reply]
I thought I could add something to my compile_commands.json and that would be enough. I created it manually and now it is empty.

**Comment by i-ky** [comment, marked as answer]
Add to your [clangd config file](https://clangd.llvm.org/config):
```yaml
If:
  PathMatch: .*\.mq4
CompileFlags:
  Add:
  - '-x'
  - 'c++'
```

**Reply by Ruslan618** [reply]
Thak you very much it helped. And how me to add additional format .mqh so that both .mq4 and .mqh were available.

**Reply by i-ky** [reply]
I guess you need `PathMatch: [.*\.mq4, .*\.mqh]` then.

**Reply by Ruslan618** [reply]
It works! Thanks again.

**Reply by Ruslan618** [reply]
Write more, please, how it should look like in the compile_commands.json file.

**Reply by i-ky** [reply]
Sorry, I don't understand your question. Basically, there should be the same `-x c++` options for every *.mq4/*.mqh file, but you probably shouldn't modify your `compile_commands.json` by hand (unless you really-really know what you are doing) and instead should modify your build system as @HighCommander4 [suggests](https://github.com/clangd/clangd/discussions/1273#discussioncomment-3532948). I have no idea, how MQL is usually compiled, so I can't provide you further hints.

**Reply by Ruslan618** [reply]
Well, just in case, I wanted to check the possibility of launching through the compile_commands.json file. The main thing is that it works. Thank you.
